FWCore/ParameterSet/python/Config.py

```diff
@@ -554,6 +558,41 @@ def extend(self,other,items=()):
                 newSeq._place(name,self)
         self.__dict__['_Process__InExtendCall'] = False
 
+    # FIXME define explicitly if the items should be referenced or copied
```

Comment:
This was, if I understood it correctly, a request from David, and it kind
of makes sense to me:
  - any customisation done on the fragment affects only the modules and
paths in the fragment itself
  - loading the fragment only adds new modules and paths to the main process

In fact if I understood the code properly, the existing Process.load()
refuses to overwrite existing objects (but I haven't tested it).

So, in our customisation we are able to do things like removing modules,
iterating over all modules and making global changes, etc. - and be
confident that it will only affect the HLT menu and not the other paths and
modules that might be present in the main process.

If we do want to affect the whole process we can apply the customisation to
it instead.

The one part I'm not sure about are the Eras customisations: I've tried to
mimic how Process.load() handles them, but I don't know how they are
supposed to work.
On 19 Mar 2015 08:23, "Martin Grunewald" <notifications@github.com> wrote:

> In FWCore/ParameterSet/python/Config.py
> <https://github.com/cms-sw/cmssw/pull/8389#discussion_r26733658>:
>
> > @@ -554,6 +558,41 @@ def extend(self,other,items=()):
> >                  newSeq._place(name,self)
> >          self.__dict__['_Process__InExtendCall'] = False
> >
> > +    # FIXME define explicitly if the items should be referenced or copied
>
> It looks to me this can add and modify, but not delete an object:
> if in "fragment" and object gets deleted, it is simply not there anymore.
> How will "process" see this? You can not just delete in "process"
> anything which is not in "fragment". Perhaps one needs in "fragment"
> and explicit list of objects to be deleted, and run through it here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8389/files#r26733658>.
>